web/static/js/components/thesis-editor.js

```diff
@@ -170,6 +178,29 @@ class ThesisEditor extends React.Component {
     })
   }
 
+  importData (page, contents) {
+    // apply page settigns
+    this.setState({
+      importContentQueueCount: contents.length,
+      importContentCompletedCount: 0,
+      title: page.title,
+      description: page.description,
+      pageModified: true
+    })
+
+    // apply page contents
+    for (let i = 0; i < contents.length; i++) {
+      const content = contents[i]
+      this.importContent(content.content_type, content, page)
+    }
```

Comment:
@jamonholmgren - I merely copied it from a different function for consistency sake.